modules/game/src/main/BSONHandlers.scala

```diff
@@ -555,6 +576,58 @@ object BSONHandlers {
 
       }
 
+      def readBackgammonGame(r: BSON.Reader): (StratGame, Metadata) = {
+
+        val gameVariant = BackgammonVariant(r intD F.variant) | BackgammonStandard
+
+        val actionStrs = NewLibStorage.OldBin.decode(GameLogic.Backgammon(), r bytesD F.oldPgn, playedPlies)
+
+        def turnUcis(turnStr: Option[String]) =
+          turnStr.map(_.split(",").toList.flatMap(backgammon.format.Uci.apply)).getOrElse(List.empty)
+
+        val backgammonGame = StratGame.Backgammon(
+          backgammon.Game(
+            situation = backgammon.Situation(
+              backgammon.Board(
+                pieces = BinaryFormat.piece
+                  .readBackgammon(r bytes F.binaryPieces, gameVariant)
+                  .filterNot { case (_, posInfo) => posInfo._2 == 0 },
+                history = backgammon.History(
+                  lastTurn = turnUcis(r strO F.historyLastTurn),
+                  currentTurn = turnUcis(r strO F.historyCurrentTurn),
+                  //we can flatten as backgammon does not have any multiaction games
+                  //TODO: Is halfMoveClock even doing anything for backgammon?
```

Comment:
i expect this calculation is wrong, but i think its wrong for many game logics. the comment about flatten is untrue as well